p2p/src/peer.rs

```diff
@@ -123,16 +126,30 @@ impl Peer {
 		*state == State::Connected
 	}
 
+	/// Unban if ban has timed out
+	pub fn recheck_ban_timeout(&self) -> bool {
+		let banned_until = self.banned_until.read().unwrap();
+	  if time::now_utc().to_timespec().sec > *banned_until {
+			let mut state = self.state.write().unwrap();
+			*state = State::Disconnected;
+			true
+		} else {
+		  false
+		}
+	}
+
 	/// Whether this peer has been banned.
 	pub fn is_banned(&self) -> bool {
 		let state = self.state.read().unwrap();
-		*state == State::Banned
+		*state == State::Banned && self.recheck_ban_timeout()
 	}
 
 	/// Set this peer status to banned
 	pub fn set_banned(&self) {
 		let mut state = self.state.write().unwrap();
 		*state = State::Banned;
+		let mut banned_until = self.banned_until.write().unwrap();
+		*banned_until = (time::now_utc() + time::Duration::minutes(42)).to_timespec().sec;
```

Comment:
Lol at 42 - lets do 3 hours initially? If this turns out to be too long then we can shorten it.
But I think we want to ban for an extended period for now.

Reply:
Maybe wrongly, I assumed we'd just ban and ban again for nodes that aren't updated. My recent restarts only banned good nodes, so maybe leave a low number at least until that issue is fixed?

For Mainnet I'm thinking a tiered slightly random approach, and repeat offenders get bumped into lower levels of oblivion, with a slower poisson process to get kicked upwards to get another chance. When DoS attacks start, I think IP-based banning won't be effective protection, and we'll need to score connected nodes by age, trying to keep a mix of old known and recently discovered nodes in the 8 spot connection pool.

Reply:
What do you mean by "banned good nodes"? Nodes are banned if they send a block that will never be valid for you. If they accepted that node, there is nothing they can do to help you.

Reply:
as per gitter chat, it seems we have examples of banning good nods incorrectly. Maybe making that painful for all the newcomers and teaching them to do rm -rf .grin/peers regularly is the right call. And mably was a bad seed node, but he already started resync and will be good again in shorter time than 3 hours.